stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -1,24 +1,33 @@
-import React, { useEffect, useState, useCallback } from "react";
+import React, { useEffect, useState, useCallback, useReducer } from "react";
 import Components from "stockflux-components";
 import { Link } from "react-router-dom";
 import "./SecuritiesTable.css";
 import * as service from "../services/SecuritiesService";
 import ValidationError from "../services/ValidationError";
-import Alert from "./Alert";
+import Alert, { AlertType } from "./Alert";
 import { TableState } from "../enums";
-import Button from "./Button";
+import ToolTip from "./ToolTip";
+import Button, { ButtonSize } from "./Button";
+import { FaPen, FaTrashAlt, FaEye, FaEyeSlash } from "react-icons/fa";
+import { MdClose, MdCheck } from "react-icons/md";
+import {
+  securitiesTableReducer,
+  initialTableState
+} from "../reducers/securitiesTableReducer";
+import PropTypes from "prop-types";
 
 const SecuritiesTable = ({ location }) => {
   const [securitiesData, setSecuritiesData] = useState([]);
-  const [messages, setMessages] = useState([]);
-  const [state, setState] = useState(TableState.LOADING);
+  const [state, dispatch] = useReducer(
+    securitiesTableReducer,
+    initialTableState
+  );
 
   const handleError = err => {
-    setState(TableState.ERROR);
     if (err instanceof ValidationError) {
-      setMessages(err.messages);
+      dispatch({ type: "error", messages: err.messages });
     } else {
-      setMessages([err.message]);
+      dispatch({ type: "error", messages: [err.message] });
```

Comment:
Also, type here being a string doesn't seem right. An enum style constant would be better. You're using `TableState.LOADING` etc. further down so I would imagine that this just slipped through the cracks.

Reply:
Yeah this just slipped through the cracks. Is changed now